magic/magic.go

```diff
@@ -0,0 +1,158 @@
+// Code generated by magicgen. DO NOT EDIT.
```

Comment:
If we auto-generate this file, perhaps we don't need to include it in the repository. We could also add it in `.gitignore`.

Reply:
I don't think we need to generate it every time we build. The only time we'll have to re-generate will be if the original `magic.h` in the criu repo changes. Do we add something to the parent makefile to generate this when `make` is invoked?

Reply:
Auto-generated files are usually excluded from the git version control.

If you think there is a reason to keep track of the changes in this file, we need to fix the following two problems:

1. Every time we run `make -C scripts/` the content of the `magic/magic.go` file changes
2. Running `make clean -C scripts/`  would remove `magic/magic.go`

Reply:
Makes sense, let's exclude it